source/modulo_components/modulo_components/component_interface.py

```diff
@@ -337,3 +339,30 @@ def lookup_transform(self, frame_name: str, reference_frame_name="world", time_p
             return result
         except tf2_py.TransformException as e:
             raise LookupTransformError(f"Failed to lookup transform: {e}")
+
+    def add_periodic_callback(self, name: str, callback: Callable[[], None]):
+        """
+        Add a periodic callback function. The provided function is evaluated periodically at the component step period.
+
+        :param name: The name of the callback
+        :param callback: The callback function that is evaluated periodically
+        """
+        if not name:
+            self.get_logger().error("Failed to add periodic function: Provide a non empty string as a name.")
+            return
+        if name in self._periodic_callbacks.keys():
+            self.get_logger().debug(f"Periodic function '{name}' already exists, overwriting.")
```

Comment:
I would put this at info level or even maybe warn (I let you decide). It seems too important to notify to keep a debug log.

Reply:
done :+1: